sdk/identity/azure-identity/test/ut/azure_cli_credential_test.cpp

```diff
@@ -359,4 +359,54 @@ TEST(AzureCliCredential, StrictIso8601TimeFormat)
       token.ExpiresOn,
       DateTime::Parse("2022-08-24T00:43:08.000000Z", DateTime::DateFormat::Rfc3339));
 }
+
+TEST(AzureCliCredential, Diagnosability)
+{
+  {
+    AzureCliTestCredential const azCliCred(
+        EchoCommand("'az' is not recognized as an internal or external command, "
+                    "operable program or batch file."));
+
+    TokenRequestContext trc;
+    trc.Scopes.push_back("https://storage.azure.com/.default");
+    try
+    {
+      static_cast<void>(azCliCred.GetToken(trc, {}));
+    }
+    catch (AuthenticationException const& e)
+    {
+      std::string const expectedMsgStart
+          = "AzureCliCredential didn't get the token: "
+            "\"'az' is not recognized as an internal or external command, "
+            "operable program or batch file.";
+
+      std::string actualMsgStart = e.what();
+      actualMsgStart.resize(expectedMsgStart.length());
+
+      // It is enough to compare StartsWith() and not deal with
+      // the entire string due to '/n' and '/r/n' differences.
+      EXPECT_EQ(actualMsgStart, expectedMsgStart);
```

Comment:
You could also just use `actualMsgStart.find(expectedMsgStart)==0` to determine if `actualMsgStart` is a prefix of `expectedMsgStart`.

Just another way of doing the same thing. The diagnostics from `EXPECT_EQ` are much better comparing two strings than comparing a `.find` and `0`.

Reply:
You're right, I think the error message is better when comparing two strings:
```
error: Expected equality of these values:
  actualMsgStart
    Which is: "AzureCliCredential didn't get the token: \"'az' is not recognized as an internal or external command, operable program or batch file"
  expectedMsgStart
    Which is: "AzureCliCredential didn't get the token: \"'az' is not recognized as an internal or external command, operable program or batch file."
```